src/runtime_src/hip/api/hip_memory.cpp

```diff
@@ -225,7 +241,7 @@ namespace xrt::core::hip
    
     auto s_hdl = hip_stream.get();
     auto cmd_hdl = insert_in_map(command_cache,
-                                std::make_shared<copy_buffer>(hip_stream, XCL_BO_SYNC_BO_TO_DEVICE, hip_mem_dst, src, size, offset));
+                                 std::make_shared<memcpy_command>(hip_stream, dst, src, size, hipMemcpyHostToDevice));
```

Comment:
shared vs unique?  Some comments would be nice.

Reply:
my understanding is that the ptr to a xrt::core::hip::command object could be shared between global command_cache and stream::m_top_event::m_chain_of_commands of a stream object. So shared_ptr is chosen for the storage type.

Comment has been added to the code.